examples/inception_v1_preprocessing_steps.ipynb

```diff
@@ -0,0 +1,327 @@
+    "response = requests.get(img_url)\n",
+    "%matplotlib inline\n",
+    "img = PIL.Image.open(BytesIO(response.content))\n",
+    "imshow(np.asarray(img))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# for getting CoreML predictions we directly pass in the PIL image after resizing\n",
+    "import coremltools\n",
+    "img = img.resize([224,224], PIL.Image.ANTIALIAS)\n",
+    "coreml_inputs = {'input__0': img}\n",
+    "coreml_output = coreml_model.predict(coreml_inputs, useCPUOnly=True)\n",
+    "coreml_pred_dict = coreml_output['InceptionV1__Logits__Predictions__Softmax__0']\n",
+    "coreml_predicted_class_label = coreml_output['classLabel']\n",
+    "\n",
+    "#for getting TF predciton we get the numpy array of the image\n",
+    "img_np = np.array(img).astype(np.float32)\n",
+    "print 'image shape:', img_np.shape\n",
+    "print 'first few values: ', img_np.flatten()[0:4], 'max value: ', np.amax(img_np)\n",
+    "img_tf = np.expand_dims(img_np, axis = 0) #now shape is [1,224,224,3] as required by TF\n",
+    "\n",
+    "# Evaluate TF and get the highest label \n",
+    "tf_input_name = 'input:0'\n",
+    "tf_output_name = 'InceptionV1/Logits/Predictions/Softmax:0'\n",
+    "with tf.Session(graph = g) as sess:\n",
+    "    tf_out = sess.run(tf_output_name, \n",
+    "                      feed_dict={tf_input_name: img_tf})\n",
+    "tf_out = tf_out.flatten()    \n",
+    "idx = np.argmax(tf_out)\n",
+    "label_file = 'imagenet_slim_labels.txt' \n",
+    "with open(label_file) as f:\n",
+    "    labels = f.readlines()\n",
+    "    \n",
+    "#print preditions   \n",
+    "print('\\n')\n",
+    "print(\"CoreML prediction class = {}, probabiltiy = {}\".format(coreml_predicted_class_label,\n",
+    "                                            str(coreml_pred_dict[coreml_predicted_class_label])))  \n",
+    "print(\"TF prediction class = {}, probability = {}\".format(labels[idx],\n",
+    "                                            str(tf_out[idx])))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Both the predicitons match, this means that the conversion was correct. However, the class label seems incorrect. What could be the reason? The answer is that we did not preprocess the image correctly before passing it to the neural network!! This is always a crucial step when using neural networks on images.\n",
+    "\n",
+    "How do we know what preprocessing to apply? This can be tricky to find sometimes. The approach is to find the source of the pre-trained model and check for the preprocessing that the author of the model used while training and evalaution. In this case, the TF model comes from the SLIM library so we find the preprocessing steps here: https://github.com/tensorflow/models/blob/edb6ed22a801665946c63d650ab9a0b23d98e1b1/research/slim/preprocessing/inception_preprocessing.py#L243\n",
```

Comment:
This paragraph doesn't render properly when displayed on screen. The problem might be due to a very long url that forces the text to go beyond border on the right. I recommend we use markdown syntax for this url.

Reply:
Yes. I'll use the markdown url syntax.